src/gfx/pal_spec.cpp

```diff
@@ -595,3 +616,61 @@ void parseExternalPalSpec(char const *arg) {
 
 	std::get<1> (*iter)(file);
 }
+
+void parseDmgPalSpec(char const * const rawArg) {
+	// Two hex digit DMG palette spec
+
+	std::string_view arg(rawArg);
+
+	if (arg.length() != 2
+	    || arg.find_first_not_of("0123456789ABCDEFabcdef"sv) != std::string_view::npos) {
+		error("Unknown DMG palette specification \"%s\"", rawArg);
+		return;
+	}
+
+	options.palSpecDmg = toHex(arg[0], arg[1]);
+
+	// Map gray shades to their DMG color indexes for fast lookup by `Rgba::grayIndex`
+	for (uint8_t i = 0; i < 4; ++i) {
+		options.dmgColors[options.dmgValue(i)] = i;
+	}
+
+	// Validate that DMG palette spec does not have conflicting colors
+	for (uint8_t i = 0; i < 4; ++i) {
+		for (uint8_t j = 0; j < 4; ++j) {
+			if (i != j && options.dmgValue(i) == options.dmgValue(j)) {
```

Comment:
```suggestion
	for (uint8_t i = 0; i < 4; ++i) {
		for (uint8_t j = i + 1; j < 4; ++j) {
			if (options.dmgValue(i) == options.dmgValue(j)) {
```
This should work because equality is reflexive.